src/components/BarChart/index.jsx

```diff
@@ -0,0 +1,109 @@
+import React, { useEffect, useState } from "react";
+import { Bar } from "react-chartjs-2";
+import { generateRandomColors, createColorsWithOpacity } from "../../shared/utils";
+
+const CHART_COUNT = 8;
+
+const BarChart = ({ data }) => {
+    const colors = generateRandomColors(CHART_COUNT);
```

Comment:
move this into useEffect

Reply:
I can't, I use it there and it won't be visible in the useEffect

Reply:
its incorrect to do such requests inside component. Use effect were created for that purpose. Move it from here

Reply:
listen, I use the line below, how do I get it from the effects later

Reply:
I won't be telling you what code to write. Think by yourself

Reply:
I don't understand why simple actions should be in effects, it's just a function call and that's it, you can explain it just so that I don't understand, and without harsh words

Reply:
https://www.reddit.com/r/reactjs/comments/8avfej/what_does_side_effects_mean_in_react/

Reply:
https://dmitripavlutin.com/react-useeffect-explanation/

Reply:
thanks for the article, the info was helpful, fixed that too